flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphSchedulingTest.java

```diff
@@ -1,637 +0,0 @@
-		}
-
-		// no deployment calls must have happened
-		assertThat(taskManager.getSubmitTaskCount(), is(0));
-
-		// all completed futures must have been returns
-		for (int i = 0; i < parallelism; i += 2) {
-			assertFalse(sourceSlots[i].isAlive());
-			assertFalse(targetSlots[i].isAlive());
-		}
-	}
-
-	/**
-	 * This tests makes sure that with eager scheduling no task is deployed if a single
-	 * slot allocation fails. Moreover we check that allocated slots will be returned.
-	 */
-	@Test
-	public void testEagerSchedulingWithSlotTimeout() throws Exception {
-
-		//  we construct a simple graph:    (task)
-
-		final int parallelism = 3;
-
-		final JobVertex vertex = new JobVertex("task");
-		vertex.setParallelism(parallelism);
-		vertex.setInvokableClass(NoOpInvokable.class);
-
-		final JobID jobId = new JobID();
-		final JobGraph jobGraph = new JobGraph(jobId, "test", vertex);
-		jobGraph.setScheduleMode(ScheduleMode.EAGER);
-
-		final BlockingQueue<AllocationID> returnedSlots = new ArrayBlockingQueue<>(2);
-		final TestingSlotOwner slotOwner = new TestingSlotOwner();
-		slotOwner.setReturnAllocatedSlotConsumer(
-			(LogicalSlot logicalSlot) -> returnedSlots.offer(logicalSlot.getAllocationId()));
-
-		final InteractionsCountingTaskManagerGateway taskManager = createTaskManager();
-		final LogicalSlot[] slots = new LogicalSlot[parallelism];
-		@SuppressWarnings({"unchecked", "rawtypes"})
-		final CompletableFuture<LogicalSlot>[] slotFutures = new CompletableFuture[parallelism];
-
-		for (int i = 0; i < parallelism; i++) {
-			slots[i] = createSingleLogicalSlot(slotOwner, taskManager, new SlotRequestId());
-			slotFutures[i] = new CompletableFuture<>();
-		}
-
-		ProgrammedSlotProvider slotProvider = new ProgrammedSlotProvider(parallelism);
-		slotProvider.addSlots(vertex.getID(), slotFutures);
-
-		final ExecutionGraph eg = createExecutionGraph(jobGraph, slotProvider);
-
-		//  we complete one future
-		slotFutures[1].complete(slots[1]);
-
-		//  kick off the scheduling
-		eg.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-		eg.scheduleForExecution();
-
-		//  we complete another future
-		slotFutures[2].complete(slots[2]);
-
-		// check that the ExecutionGraph is not terminated yet
-		assertThat(eg.getTerminationFuture().isDone(), is(false));
-
-		// time out one of the slot futures
-		slotFutures[0].completeExceptionally(new TimeoutException("Test time out"));
-
-		assertThat(eg.getTerminationFuture().get(), is(JobStatus.FAILED));
-
-		// wait until all slots are back
-		for (int i = 0; i < parallelism - 1; i++) {
-			returnedSlots.poll(2000, TimeUnit.MILLISECONDS);
-		}
-
-		//  verify that no deployments have happened
-		assertThat(taskManager.getSubmitTaskCount(), is(0));
-	}
-
-	/**
-	 * Tests that an ongoing scheduling operation does not fail the {@link ExecutionGraph}
-	 * if it gets concurrently cancelled.
-	 */
-	@Test
-	public void testSchedulingOperationCancellationWhenCancel() throws Exception {
```

Comment:
I think the implementation is a bit different in `DefaultScheduler` because the job canceling will increment mod version of all vertices, and further slot assignment and deployment will be skipped as a result and the job status will not be affected. The version incrementing is tested via `cancelJobWillIncrementVertexVersions`. `releaseSlotIfVertexVersionOutdated` and `skipDeploymentIfVertexVersionOutdated` test the skipped slot assignment and deployment.

Reply:
I think you are right. Marking this conversation as resolved.